scipy/spatial/transform/rotation.py

```diff
@@ -657,3 +656,98 @@ def __mul__(self, other):
         if self._single and other._single:
             result = result[0]
         return self.__class__(result, normalized=True)
+
+
+class Slerp(object):
+    """Spherical Linear Interpolation of Rotations.
+
+    The SLERP algorithm guarantees that the interpolated rotations will have
+    constanct angular velocity between each successive "keyframe" rotation.
+
+    Given a set of `rotations` and corresponding fixed `timestamps`, this class
+    will initialize a `Slerp` instance which is an interpolator. The
+    interpolator can be called with another set of timestamps to return a
+    single `Rotation` object containing the interpolated rotations.
+
+    Parameters
+    ----------
+    rotations : a `Rotation` instance
+        This object contains the fixed rotations between which the
+        interpolation will be done.
+    timestamps : array_like, shape (N,)
+        An array of timestamps, sorted in increasing order, for the fixed
+        rotations. The number of times must be equal to the number of rotations
+        specified in the `rotations` object.
+
+    Methods
+    -------
+    __call__
+    """
+    def __init__(self, rotations, timestamps):
+        self._rots = rotations
+        # Possibly sort timestamps, or add a sorted flag?
+        self._times = np.asarray(timestamps)
+
+        if self._times.shape[0] != self._rots._quat.shape[0]:
+            raise ValueError("Expected number of rotations to be equal to "
+                             "number of keyframe timestamps, got {} rotations "
+                             "and {} timestamps.".format(
+                                self._rots._quat.shape[0],
+                                self._times.shape[0],
+                             ))
+
+    def __call__(self, timestamps):
+        """Generates the interpolated rotation at given times.
```

Comment:
Use imperatives, "Generate ...". And generally try to improve formulations as you go through them. I mean read it, if you feel that something is not quite right with it --- try to rethink and rephrase that.